### PR TITLE
Replace samples/vision_inference with a Python script + README.

### PR DESCRIPTION
**Pull request by ScottTodd**
Also remove image_util and third_party/stblib, which were only used for the C code in that sample.

Progress on https://github.com/google/iree/issues/9374

### CODE REVIEW

**Comment by benvanik** [comment]
🤤

**Review by benvanik — APPROVED** [review]

File: samples/vision_inference/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# Run the program, passing the path to the binary file as a function input.
+iree-run-module \
+  /tmp/mnist_test.bin \
+  --function_input=1x28x28x1xf32=@/tmp/mnist_test.bin
+
+# Observe the results - a list of prediction confidence scores for each digit.
+```
+
+<!-- TODO(scotttodd): lit test for that ^ (requires python in lit.cfg.py) -->
```
Comment (ScottTodd) [inline samples/vision_inference/README.md:29]: I spent a good 30 minutes trying to figure out how to copy the magic from
https://github.com/google/iree/blob/74d741e62c0646ddd34a927ace594b8829858d48/llvm-external-projects/iree-dialects/test/lit.cfg.py#L65-L70 https://github.com/google/iree/blob/74d741e62c0646ddd34a927ace594b8829858d48/llvm-external-projects/iree-dialects/test/CMakeLists.txt#L5-L10 https://github.com/google/iree/blob/74d741e62c0646ddd34a927ace594b8829858d48/llvm-external-projects/iree-dialects/test/lit.site.cfg.py.in#L15
into https://github.com/google/iree/blob/main/samples/lit.cfg.py, but had no luck (`llvm_config` was always `None`). Could be Windows / Python making it complicated... idk.

Comment (GMNGeoffrey) [reply samples/vision_inference/README.md:29]: `llvm_config` is from using llvm-lit, which isn't the same as lit, which is... cool

Comment (ScottTodd) [reply samples/vision_inference/README.md:29]: Ah, so would we need to swap out `${LLVM_SOURCE_DIR}/utils/lit/lit.py` in `iree_lit_test.cmake`? https://github.com/google/iree/blob/d97e8bc0d306568ea047f408610779f7450f9230/build_tools/cmake/iree_lit_test.cmake#L67-L78
Could we just always use `llvm-lit`?

Comment (GMNGeoffrey) [reply samples/vision_inference/README.md:29]: llvm-lit is weird and I don't really understand it. You're welcome to look into this, but it's a whole can of worms (don't forget about Bazel)

Comment (ScottTodd) [reply samples/vision_inference/README.md:29]: I just want to run a python script in a test 😭 (well, `iree-compile` -> python script -> `iree-run-module`). I'll punt on test coverage for now I guess...

**Pull request merged**